### PR TITLE
CNS-503 cleanup unit tests (part 1)

### PR DESCRIPTION
**Pull request by orenl-lava**

### CODE REVIEW

File: x/subscription/keeper/subscription_test.go [hunk 1]
```diff
@@ -242,253 +168,210 @@ func TestCreateSubscription(t *testing.T) {
 }
 
 func TestSubscriptionExpiration(t *testing.T) {
-	ts := setupTestStruct(t, 1)
-	keeper := ts.keepers.Subscription
+	ts := newTester(t)
+	ts.SetupAccounts(1, 0, 0) // 2 sub, 0 adm, 0 dev
```
Comment (oren-lava) [inline x/subscription/keeper/subscription_test.go:172]: comment: 2 sub -> 1 sub

File: x/subscription/keeper/subscription_test.go [hunk 2]
```diff
@@ -242,253 +168,210 @@ func TestCreateSubscription(t *testing.T) {
...
-
-	err = keeper.CreateSubscription(ts.ctx, creator, creator, ts.plans[0].Index, 1)
-	require.Nil(t, err)
-
-	_, found := keeper.GetSubscription(ts.ctx, creator)
+	_, found := ts.getSubscription(sub1Addr)
 	require.True(t, found)
 
-	timestamp := ts.ctx.BlockTime()
+	// advance 1 month + epoch, subscriptiokn should expire
```
Comment (oren-lava) [inline x/subscription/keeper/subscription_test.go:182]: nit: subscription

File: x/subscription/keeper/subscription_test.go [hunk 3]
```diff
@@ -242,253 +168,210 @@ func TestCreateSubscription(t *testing.T) {
...
+	_, adm1Addr := ts.Account("adm1")
+	_, dev1Addr := ts.Account("dev1")
+	plan := ts.Plan("mock")
 
-	blocksToSave, err := ts.keepers.Epochstorage.BlocksToSave(ts.ctx, uint64(ts.ctx.BlockHeight()))
+	_, err := ts.TxSubscriptionBuy(sub1Addr, sub1Addr, plan.Index, 3)
 	require.Nil(t, err)
 
-	// fill memory
-	for uint64(ts.ctx.BlockHeight()) < blocksToSave {
```
Comment (oren-lava) [inline x/subscription/keeper/subscription_test.go:380]: is this really not needed for the test?

**Review by oren-lava — CHANGES_REQUESTED** [review]

File: x/subscription/keeper/subscription_test.go (hunk 3 above)
Comment (orenl-lava) [reply x/subscription/keeper/subscription_test.go:380]: the tests pass with and without it. 
I actually do not understand _why_ it was needed in the first place.
the `NewTester()` function kick-starts the blocks with a default AdvanceEpoch(). 
so the question is: is this really needed for any tests?

File: x/plans/keeper/plan_test.go [hunk 4]
```diff
@@ -1,70 +1,64 @@
...
 
-func createNPlanEntry(keeper *keeper.Keeper, ctx sdk.Context, n int) []types.Plan {
-	items := make([]types.Plan, n)
-	for i := range items {
-		items[i].Index = strconv.Itoa(i)
-
-		keeper.AddPlan(ctx, items[i])
-	}
-	return items
+func (ts *tester) findPlan(index string, block uint64) (types.Plan, bool) {
```
[the remark this answers is not in the document]
Comment (orenl-lava) [reply x/plans/keeper/plan_test.go]: fixed: 
9b111561
0e134976
6eea644a

File: x/plans/keeper/plan_test.go [hunk 5]
```diff
@@ -187,244 +154,186 @@ func TestInvalidPlanAddition(t *testing.T) {
...
-	ts.advanceEpochUntilStale()
+	ts.AdvanceEpoch()
 
-	// create an additional plan and add it to the store to trigger plan deletion code
-	newPlan := testPlans[1]
-	newPlan.OveruseRate += 20
-	newPlanBlockHeight := uint64(sdk.UnwrapSDKContext(ts.ctx).BlockHeight())
-	newPlan.Block = newPlanBlockHeight
-	err = ts.keepers.Plans.AddPlan(sdk.UnwrapSDKContext(ts.ctx), newPlan)
+	// add 3rd plan
```
[the remark this answers is not in the document]
Comment (orenl-lava) [reply x/plans/keeper/plan_test.go]: fixed; 8acb1958

File: x/projects/keeper/project_test.go [hunk 6]
```diff
@@ -97,310 +62,319 @@ func (ts *testStruct) prepareData(numSub, numAdmin, numDevel int) {
...
 	}
 
 	for _, tt := range templates {
-		ts.projects[tt.code] = types.ProjectData{
+		pd := types.ProjectData{
 			Name:        tt.name,
 			Enabled:     tt.enabled,
 			ProjectKeys: tt.keys,
-			Policy:      tt.policy,
+			Policy:      &policy,
```
[the remark this answers is not in the document]
Comment (orenl-lava) [reply x/projects/keeper/project_test.go]: fixed: 8acb1958

File: x/projects/keeper/project_test.go [hunk 7]
```diff
@@ -97,310 +62,319 @@ func (ts *testStruct) prepareData(numSub, numAdmin, numDevel int) {
...
 
+func (ts *tester) getProjectForBlock(projectID string, block uint64) (types.Project, error) {
+	return ts.Keepers.Projects.GetProjectForBlock(ts.Ctx, projectID, block)
+}
+
+func (ts *tester) getProjectForDeveloper(devkey string) (types.Project, error) {
+	return ts.Keepers.Projects.GetProjectForDeveloper(ts.Ctx, devkey, ts.BlockHeight())
+}
+
+func (ts *tester) queryDeveloper(devkey string) (*types.QueryDeveloperResponse, error) {
```
[the remark this answers is not in the document]
Comment (orenl-lava) [reply x/projects/keeper/project_test.go]: fixed: e7e5ca31

File: x/projects/keeper/project_test.go [hunk 8]
```diff
@@ -97,310 +62,319 @@ func (ts *testStruct) prepareData(numSub, numAdmin, numDevel int) {
...
 }
 
 func TestProjectsServerAPI(t *testing.T) {
-	ts := newTestStruct(t)
-	ts.prepareData(2, 1, 5) // 2 sub, 1 adm, 5 dev
-
-	sub1Acc := common.CreateNewAccount(ts._ctx, *ts.keepers, 20000)
-	sub1Addr := sub1Acc.Addr.String()
+	ts := newTester(t)
+	ts.SetupAccounts(1, 0, 2) // 0 sub, 0 adm, 2 dev
```
[the remark this answers is not in the document]
Comment (orenl-lava) [reply x/projects/keeper/project_test.go]: fixed: 8acb1958

File: x/subscription/keeper/subscription_test.go (hunk 1 above)
Comment (orenl-lava) [reply x/subscription/keeper/subscription_test.go:168]: fixed: 8acb1958

File: x/subscription/keeper/subscription_test.go (hunk 2 above)
Comment (orenl-lava) [reply x/subscription/keeper/subscription_test.go]: fixed: d72f0c59

File: x/subscription/keeper/subscription_test.go (hunk 3 above)
Comment (oren-lava) [reply x/subscription/keeper/subscription_test.go:380]: I'm not sure either. It's @Yaroms code, we can ask him. Maybe something is wrong with the test

File: testutil/common/tester.go [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+	projectstypes "github.com/lavanet/lava/x/projects/types"
+	spectypes "github.com/lavanet/lava/x/spec/types"
+	subscriptionkeeper "github.com/lavanet/lava/x/subscription/keeper"
+	subscriptiontypes "github.com/lavanet/lava/x/subscription/types"
+)
+
+type Tester struct {
+	T *testing.T
+
+	GoCtx   context.Context
```
Comment (Yaroms) [inline testutil/common/tester.go:22]: are the 2 contexts synchronized?

Comment (orenl-lava) [reply testutil/common/tester.go:22]: yes: all the Advance{Something})() methods update both of them.

**Review by Yaroms — APPROVED** [review]

**Pull request merged**